### PR TITLE
Allow the engine to use a native configuration

### PR DESCRIPTION
**Pull request by johannegger**
-- in the case of scalastyle a configuration is mandatory.
so the engine can either be called with codacy-parameters (as before)
or it can contain a file 'scalastyle_config.xml' or 'scalastyle-config.xml' in the project root (native)
or the default configuration will be used

### CODE REVIEW

File: src/main/scala/codacy/scalastyle/ScalaStyle.scala [hunk 1]
```diff
@@ -32,7 +40,16 @@ object ScalaStyle extends Tool {
     }
   }
 
-  def parseToolResult(lines: List[String], path: Path): List[Result] = {
+  private lazy val configFileName = "scalastyle_config.xml"
+  private lazy val configFileNames = List(configFileName, "scalastyle-config.xml")
```
Comment (machadoit) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: Am I missing something? Or this is a list with the name of the config 2 times?

File: src/main/scala/codacy/scalastyle/ScalaStyle.scala [hunk 2]
```diff
@@ -3,24 +3,32 @@ package codacy.scalastyle
...
       val fullConf = ToolHelper.getPatternsToLint(conf)
       val filesToLint: List[String] = files.fold(List(path.toString)) {
         paths =>
           paths.map(_.toString).toList
       }
 
-      val configuration = List("--config", getConfigFile(fullConf).getAbsolutePath.toString)
+      val configuration = List("--config",
+        //priorities: codacy-patterns then a native config in the project-root then the default config
+        getConfigFile(fullConf).orElse(nativeConfigFile).getOrElse(defaultConfigFile).getAbsolutePath
```
Comment (machadoit) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: o fullConf é antes do nativ?

File: src/main/scala/codacy/scalastyle/ScalaStyle.scala (hunk 1 above)
Comment (johannegger) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: '-' vs '_' ;)

File: src/main/scala/codacy/scalastyle/ScalaStyle.scala (hunk 2 above)
Comment (johannegger) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: sim full-conf é codacy-patterns

File: src/main/scala/codacy/scalastyle/ScalaStyle.scala (hunk 1 above)
Comment (machadoit) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: oops ;)

File: src/main/scala/codacy/scalastyle/ScalaStyle.scala [hunk 3]
```diff
@@ -32,7 +40,16 @@ object ScalaStyle extends Tool {
     }
   }
 
-  def parseToolResult(lines: List[String], path: Path): List[Result] = {
+  private lazy val configFileName = "scalastyle_config.xml"
+  private lazy val configFileNames = List(configFileName, "scalastyle-config.xml")
+
+  private lazy val defaultConfigFile: File = {
+    (better.files.File.root / "docs" / configFileName).toJava
```
Comment (rtfpessoa) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: Não existe um path para resources?

Comment (johannegger) [inline src/main/scala/codacy/scalastyle/ScalaStyle.scala]: no, the file resides in the docker root. do you think it's better to use the one in the resources?

**Pull request merged**